### DISCUSSION TITLE
streams-i2s-webserver_wav.ino - white noise in right channel at sample rate of 32000Hz

### DISCUSSION THREAD

**Discussion by mmcc-xx** [pschatzmann/arduino-audio-tools#834] [Q&A] [answered]
I'm continuing to try to find a way to stream 44.1KHz sampled audio. My input device is an I2S microphone.

The streams-i2s-webserver_wav.ino sample works fine for me at sample rates of 22050KHz and 24000HKz. I'm getting audio in the left channel, and the right channel is silent. I assume that is how it is supposed to be since I have the L/R pin on the mic going to ground.

I have `ConverterFillLeftAndRight<int16_t> filler(LeftIsEmpty);` though the results seem to be the same if I have `ConverterFillLeftAndRight<int16_t> filler(RightIsEmpty);` or even if I leave the filler out of the server.begin statement

If I set the sample rate to 32000Hz or 44100KHz, I get the expected audio in the left channel, but what sounds like white noise in the right channel. The image below shows the waveform in Audacity - you can see that the left channel contains an audio signal. Note that the right channel values are all below 0.

I'd attach the wav but that isn't permitted.

Any idea what is happening here?

![image](https://github.com/pschatzmann/arduino-audio-tools/assets/1315341/0d1a9a06-1491-49a5-954d-ebb2d6724689)

**Comment by pschatzmann** [comment, marked as answer]
If you have only one channel, why don't you just set channels = 1 ?

**Reply by mmcc-xx** [reply]
Well I feel like a dope. I assumed that with an I2S stream you had to use 2 channels because it was set to 2 channels in all of the examples, and the filler was some way to fill in the channel that didn't have any data.
 
1 channel, 32000Hz, no filler worked just fine.
1 channel, 44100Hz, no filler white noise. In audacity, it looks like the signal is in there, but drowned out by the noise:

![image](https://github.com/pschatzmann/arduino-audio-tools/assets/1315341/305a7d3f-ad01-40a6-896c-1d618b1e5bcd)
